src/main/java/ch/njol/skript/expressions/ExprExperience.java

```diff
@@ -90,58 +94,58 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
 				return new Class[] {Experience[].class, Number[].class};
 			case SET:
 				return new Class[] {Experience.class, Number.class};
-			case RESET:
+			default:
 				return null;
 		}
-		return null;
 	}
 	
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
-		double eventExp;
-		if (e instanceof ExperienceSpawnEvent) {
-			eventExp = ((ExperienceSpawnEvent) e).getSpawnedXP();
-		} else if (e instanceof BlockBreakEvent) {
-			eventExp = ((BlockBreakEvent) e).getExpToDrop();
-		} else if (e instanceof PlayerExpChangeEvent) {
-			eventExp = ((PlayerExpChangeEvent) e).getAmount();
-		} else {
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		int experience;
+		if (event instanceof ExperienceSpawnEvent)
+			experience = ((ExperienceSpawnEvent) event).getSpawnedXP();
+		else if (event instanceof BlockBreakEvent)
+			experience = ((BlockBreakEvent) event).getExpToDrop();
+		else if (event instanceof PlayerFishEvent)
+			experience = ((PlayerFishEvent) event).getExpToDrop();
+		else if (event instanceof PlayerExpChangeEvent)
+			experience = ((PlayerExpChangeEvent) event).getAmount();
+		else
 			return;
-		}
-		if (delta == null) {
-			eventExp = 0;
-		} else {
-			for (Object obj : delta) {
-				double value = obj instanceof Experience ? ((Experience) obj).getXP() : ((Number) obj).doubleValue();
+
+		if (delta != null) {
+			for (Object object : delta) {
+				int value = object instanceof Experience ? ((Experience) object).getXP() : ((Number) object).intValue();
 				switch (mode) {
 					case ADD:
-						eventExp += value;
+						experience += value;
 						break;
 					case SET:
-						eventExp = value;
+						experience = value;
 						break;
 					case REMOVE:
 					case REMOVE_ALL:
-						eventExp -= value;
+						experience -= value;
 						break;
 					case RESET:
 					case DELETE:
 						assert false;
 						break;
 				}
 			}
+		} else {
+			experience = 0;
 		}
-
 		
-		eventExp = Math.max(0, Math.round(eventExp));
-		int roundedEventExp = (int) eventExp;
-		if (e instanceof ExperienceSpawnEvent) {
-			((ExperienceSpawnEvent) e).setSpawnedXP(roundedEventExp);
-		} else if (e instanceof BlockBreakEvent) {
-			((BlockBreakEvent) e).setExpToDrop(roundedEventExp);
-		} else if (e instanceof PlayerExpChangeEvent) {
-			((PlayerExpChangeEvent) e).setAmount(roundedEventExp);
-		}
+		experience = Math.max(0, Math.round(experience));
```

Comment:
`experience` is an integer now.
```suggestion
		experience = Math.max(0, experience);
```